src/main/java/org/datadog/jmxfetch/Instance.java

```diff
@@ -157,6 +175,7 @@ public Instance(
                 emptyDefaultHostnameObj != null ? (Boolean) emptyDefaultHostnameObj : false;
 
         this.lastCollectionTime = 0;
+        this.initialRefreshTime = 0;
```

Comment:
I don't think you need this at all, to be honest.

I would rely on `lastRefreshTime` and would do something like the following pseudocode:  
```
period = (this.lastRefreshTime==0) ?  this.initialRefreshBeansPeriod : this.refreshBeansPeriod;

if isPeriodDue(this.lastRefreshTime, period )) {
        // refresh beans
}
```

And as mentioned above `initialRefreshBeansPeriod` should default to `DEFAULT_REFRESH_BEANS_PERIOD`, so that if nothing is set there's no change in behavior.

Reply:
Per checking, `lastRefreshTime` is no longer 0 when it gets to this point since it has been updated in [refreshBeansList()](https://github.com/DataDog/jmxfetch/blob/master/src/main/java/org/datadog/jmxfetch/Instance.java#L401) called in init() to initially populate the beans list. Here is the 2nd time it's been called for the 1st _refresh_.
Still, I've updated the code inspired from this. Thanks!